File: 2016/ioi_day2/messy/pas/messy.pas
unit messy;

interface

type
    TIntArray = array of longint;
    
function restore_permutation(n, w, r : longint) : TIntArray;

implementation

uses graderlib;

function restore_permutation(n, w, r : longint) : TIntArray;
var
    res : array of longint;
begin
  add_element('0');
  compile_set();
  check_element('0');
  setlength(res, 0);
  restore_permutation := res;
end;

end.

File: 2016/ioi_day2/messy/c/messy.c
#include "messy_c.h"

void restore_permutation(int n, int w, int r, int* result) {
    add_element("0");
    compile_set();
    check_element("0");
    for (int i = 0; i < n; i++) {
        result[i] = i;
    }
}
